src/bp-members/bp-members-filters.php

```diff
@@ -830,39 +830,56 @@ function bb_repair_member_profile_links_callback( $is_background = false, $paged
 		return;
 	}
 
+	global $wpdb;
+
+	// Define default values.
 	$per_page = 50;
+	$offset   = 0;
 
-	if ( $is_background ) {
-		$offset = ( ( $paged - 1 ) * $per_page );
-	} else {
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-		$offset = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
-	}
+	if ( ! $is_background ) {
 
-	$args = array(
-		'fields' => 'ID',
-		'number' => $per_page,
-		'offset' => $offset,
-	);
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		$offset   = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
+		$per_page = 20;
+
+		// Set limit while repair the member slug.
+		$user_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT u.ID FROM `{$wpdb->prefix}users` AS u LEFT JOIN `{$wpdb->prefix}usermeta` AS um ON ( u.ID = um.user_id AND um.meta_key = %s ) WHERE ( um.user_id IS NULL OR LENGTH(meta_value) = %d ) ORDER BY u.ID ASC LIMIT %d, %d",
+				'bb_profile_slug',
+				40,
+				0,
+				$per_page
+			)
+		);
+	} else {
 
-	$user_ids = get_users( $args );
+		// Get all users who have not generate unique slug while it runs from background.
+		$user_ids = $wpdb->get_col(
```

Comment:
Why there is two diff queries for $is_background. like one is querying the users which has unique username length equal to 40 char. another is doing users of which haven't generated unique profile slug.

It's confusing doing two diff tasks in diff mode from one signal job. I am having hard time as a developer understanding the code in single shot.

Reply:
The first query runs when we run tools, and the second runs from the background while updating the plugin.
As per the technical document, I have made the changes to repair the member slug from 40 characters to 8-2 characters.
So, both will run from a single function.